### PR TITLE
Enable JSCS compatible rules

### PR DESCRIPTION
**Pull request by ruiquelhas**
Here is the compatibility table using [jscs-config-seegno](https://github.com/seegno/jscs-config-seegno) preset as reference:

JSCS rule | ESLint rule | Remarks
--------- | ----------- | -------
`disallowDanglingUnderscores` | `no-underscore-dangle` |
`disallowEmptyBlocks` | `no-empty` |
`disallowGeneratorsInDescribeFunctions` | `require-yield` | custom JSCS rule, covered by `require-yield`
`disallowKeywords` | `strict` | in `strict` mode, the parser reports the error
`disallowKeywordsInComments` | `no-warning-comments` | does not work for block comments
`disallowKeywordsOnNewLine` | None | `keyword-spacing` does not do the trick
`disallowMixedSpacesAndTabs` | `no-mixed-spaces-and-tabs` |
`disallowMultipleLineBreaks` | `no-multiple-empty-lines` |
`disallowMultipleLineStrings` | `no-multi-str` |
`disallowMultipleSpaces` | `no-multi-spaces` |
`disallowMultipleVarDecl` | `one-var` |
`disallowNewlineBeforeBlockStatements` | None | `space-before-blocks` does not do the trick
`disallowNodeTypes` | `no-labels` |
`disallowOnlyFilterInTestFunctions` | None | custom JSCS rule, covered by `eslint-plugin-mocha`
`disallowOperatorBeforeLineBreak` | `operator-linebreak` | function calls are ignored
`disallowPaddingNewlinesInBlocks` | `padded-blocks` |
`disallowQuotedKeysInObjects` | `quote-props` |
`disallowSpaceAfterObjectKeys` | `key-spacing` |
`disallowSpaceAfterPrefixUnaryOperators` | `space-unary-ops` |
`disallowSpaceBeforeComma` | `comma-spacing` |
`disallowSpaceBeforePostfixUnaryOperators` | `space-unary-ops` |
`disallowSpaceBeforeSemicolon` | `semi-spacing` |
`disallowSpacesInCallExpression` | `no-spaced-func` |
`disallowSpacesInFunctionDeclaration` | `space-before-function-paren` |
`disallowSpacesInsideArrayBrackets` | `array-bracket-spacing` |
`disallowSpacesInsideParentheses` | `space-in-parens` |
`disallowSpacesInsideParenthesizedExpression` | `space-in-parens` |
`disallowSpacesInsideTemplateStringPlaceholders` | `template-curly-spacing` |
`disallowTrailingComma` | `comma-dangle` |
`disallowTrailingWhitespace` | `no-trailing-spaces` |
`disallowYodaConditions` | `yoda` |
`esnext` | `env.es6` |
`requireBlocksOnNewline` | None |
`requireCamelCaseOrUpperCaseIdentifiers` | `id-match` | uses a regular expression
`requireCapitalizedComments` | None |
`requireCapitalizedConstructors` | `new-cap` |
`requireCommaBeforeLineBreak` | None | `comma-style`
`requireCurlyBraces` | `curly` | does not cover `try`, `catch` and `finally`
`requireDotNotation` | `dot-notation` |
`requireLineBreakAfterVariableAssignment` | None | partially covered by `one-var`
`requireLineFeedAtFileEnd` | `eol-last` |
`requireMatchingFunctionName` | None |
`requirePaddingNewLineAfterVariableDeclaration` | None |
`requirePaddingNewLinesAfterBlocks` | None | partially covered by `padded-blocks`
`requirePaddingNewLinesAfterUseStrict` | None | doesn't make sense with node `>=v6`
`requirePaddingNewLinesBeforeExport` | None |
`requirePaddingNewLinesBeforeLineComments` | None |
`requirePaddingNewlinesBeforeKeywords` | None |
`requireParenthesesAroundIIFE` | `wrap-iife` | strict invocation format
`requireSemicolons` | `semi` |
`requireShouldAssertionExecution` | None | custom JSCS rule, covered by `no-unused-expressions`
`requireSpaceBeforeBinaryOperators` | `space-infix-ops` |
`requireSpaceBeforeDestructuredValues` | None | covered by `key-spacing` 
`requireSpaceAfterComma` | `comma-spacing` |
`requireSpaceAfterKeywords` | `keyword-spacing` |
`requireSpaceAfterLineComment` | `spaced-comment` |
`requireSpaceBeforeBinaryOperators` | `space-infix-ops`|
`requireSpaceBeforeBlockStatements` | `space-before-blocks` |
`requireSpaceBeforeKeywords` | `keyword-spacing` |
`requireSpaceBeforeObjectValues` | `object-curly-spacing` |
`requireSpaceBetweenArguments`| `comma-spacing` |
`requireSpacesInAnonymousFunctionExpression` | `space-before-blocks` |
`requireSpacesInConditionalExpression` | `space-infix-ops` |
`requireSpacesInForStatement` | `semi-spacing` and `space-infix-ops` |
`requireSpacesInsideObjectBrackets` | `object-curly-spacing` |
`requireSpread` | `prefer-spread` |
`requireSqlTemplate` | None | custom JSCS rule
`requireTemplateStrings` | `prefer-template` |
`safeContextKeyword` | `consistent-this` |
`validateIndentation` | `indent` |
`validateLineBreaks` | `linebreak-style` |
`validateOrderInObjectKeys` | None | covered by `eslint-plugin-sorting`
`validateParameterSeparator` | `comma-spacing` |
`validateQuoteMarks` | `quotes` |

Effectively, the following rules (as defined by the `jscs` preset) are still not applied:
- [`disallowKeywordsOnNewLine`](http://jscs.info/rule/disallowKeywordsOnNewLine)
- [`disallowNewlineBeforeBlockStatements`](http://jscs.info/rule/disallowNewlineBeforeBlockStatements)
- [`requireBlocksOnNewline`](http://jscs.info/rule/requireBlocksOnNewline)
- [`requireCapitalizedComments`](http://jscs.info/rule/requireCapitalizedComments)
- [`requireLineBreakAfterVariableAssignment`](http://jscs.info/rule/requireLineBreakAfterVariableAssignment)
- [`requireMatchingFunctionName`](http://jscs.info/rule/requireMatchingFunctionName)
- [`requirePaddingNewLineAfterVariableDeclaration`](http://jscs.info/rule/requirePaddingNewLineAfterVariableDeclaration)
- [`requirePaddingNewLinesAfterBlocks`](http://jscs.info/rule/requirePaddingNewLinesAfterBlocks)
- [`requirePaddingNewLinesBeforeExport`](http://jscs.info/rule/requirePaddingNewLinesBeforeExport)
- [`requirePaddingNewlinesBeforeKeywords`](http://jscs.info/rule/requirePaddingNewLinesBeforeKeywords)
- [`requirePaddingNewLinesBeforeLineComments`](http://jscs.info/rule/requirePaddingNewLinesBeforeLineComments)
- [`requireSqlTemplate`](https://github.com/seegno/jscs-config-seegno#requiresqltemplate)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -46,47 +44,58 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^_$|^_[a-z]+[a-zA-Z]*$|^[A-Z][a-z]+[a-zA-Z]*$|^[A-Z]+_[A-Z]+$|^[a-z]+[A-Za-z]*$'],
```
Comment (ruiquelhas) [inline src/index.js]: This regular expression sucks, it's basically just a placeholder. We need to see what kind of variable naming convention we will use and what exactly it will allow.

**Comment by rplopes** [comment]
This looks great. Is there anything blocking this PR? If there are controversial changes or specific things we're not 100% happy with yet, can we at least merge the non-controversial, quick win ones?

**Comment by ruiquelhas** [comment]
I guess what's blocking this is the fact that some `jscs` rules don't have a direct match, and we would need to create custom `eslint` rules for them. However, I believe it should make an incremental approach, and reduce our current `jscs` preset to the rules that are not covered by this one, and then make a full migration.

**Comment by rplopes** [comment]
Agree, we don't need a one-step big bang refactor to drop jscs immediately. We can do it one step at a time, getting better at each step along the way. And there's plenty of those steps in this PR to get us to a better state than where we are now 👍

File: src/index.js [hunk 2]
```diff
@@ -122,16 +133,18 @@ module.exports = {
...
     'no-sync': 'error',
-    'no-this-before-super': 'error',
     'no-throw-literal': 'error',
     'no-trailing-spaces': 'error',
     'no-undef-init': 'error',
     'no-undefined': 'off',
-    'no-underscore-dangle': 'off',
+    'no-underscore-dangle': ['error', {
+      allow: ['_'],
+      allowAfterThis: true
```
Comment (rplopes) [inline src/index.js]: Why?

File: src/index.js [hunk 3]
```diff
@@ -122,16 +133,18 @@ module.exports = {
...
+    'no-spaced-func': 'error',
     'no-sync': 'error',
-    'no-this-before-super': 'error',
     'no-throw-literal': 'error',
     'no-trailing-spaces': 'error',
     'no-undef-init': 'error',
     'no-undefined': 'off',
-    'no-underscore-dangle': 'off',
+    'no-underscore-dangle': ['error', {
+      allow: ['_'],
```
Comment (rplopes) [inline src/index.js]: Reading the examples of correct code for this rule in the [docs](http://eslint.org/docs/rules/no-underscore-dangle), it looks like this might not be necessary. Have you tested without it?

File: src/index.js (hunk 2 above)
Comment (ruiquelhas) [inline src/index.js]: This was more like a personal choice. I'm assuming that underscores can be used as prefix for private variables and those should only be accessed directly in their scope (and have some kind of accessor for the outside), thus allowing something like `this._foo = 'bar';`.

File: src/index.js (hunk 3 above)
Comment (ruiquelhas) [inline src/index.js]: You're right, it's not needed. I've missed it when I was testing.

File: src/index.js [hunk 4]
```diff
@@ -140,41 +153,44 @@ module.exports = {
...
+      memberSyntaxSortOrder: ['none', 'all', 'single', 'multiple']
+    }],
+    'sort-vars': 'error',
+    'space-before-blocks': 'error',
+    'space-before-function-paren': ['error', 'never'],
+    'space-in-parens': 'error',
+    'space-infix-ops': 'error',
+    'space-unary-ops': ['error', {
+      nonwords: false,
+      words: false
```
Comment (rplopes) [inline src/index.js]: The jscs rule only seems to apply to nonwords. If we apply to words also, that means writing code like `typeof!foo` and `new[foo][0]`, which seems to go against our convention. Is this intentional?

Otherwise, if we don't want this rule to apply to words, we'll be using the rule's default values, and thus we can simplify these lines to `'space-unary-ops': 'error',`

File: src/index.js [hunk 5]
```diff
@@ -24,19 +24,17 @@ module.exports = {
     'block-spacing': 'off',
     'brace-style': 'off',
     camelcase: 'off',
-    'comma-dangle': 'off',
```
Comment (rplopes) [inline src/index.js]: Some rules, like this one, don't seem to have an addition changing from `off` to `error`. Any reason for that?

File: src/index.js (hunk 4 above)
Comment (ruiquelhas) [inline src/index.js]: You're right, `error` should be enough.

File: src/index.js (hunk 5 above)
Comment (ruiquelhas) [inline src/index.js]: Because they are already set on the `recommended` config that is being inherited.

Comment (ruiquelhas) [inline src/index.js]: Or they "have" the default value already.

File: src/index.js [hunk 6]
```diff
@@ -46,47 +44,58 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^[_a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z]+_[A-Z][A-Z]+$'],
```
Comment (rplopes) [inline src/index.js]: Was this regex extracted from any particular source? If it's to match camel case or upper case, I believe it can still be improved. For instance, that upper case won't accept numbers, only seems to accept one underscore and requires at least 2 letters at each side of that underscore.

Maybe replace the upper case part of the regex with something like `^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$`.

Comment (ruiquelhas) [inline src/index.js]: No, I've tested some variants, but it surely can be improved.

Your suggestion means we accept something such as:

```js
const A1_2 = 'foo';
const A1_B = 'bar';
```

which might be a bit weird. What do you think?

Comment (rplopes) [inline src/index.js]: If those examples are valid upper case notation, I think it's good to accept them in this rule (they might not be the best choice within the code, but that's a different matter). I see this enabling some better examples such as `OPTION_1` or `N_TIMES`.

Anyway, the specific regex I shared is more to illustrate the point than to be prescriptive. I think that at least supporting 0-n underscores and numbers would be important.

File: src/index.js (hunk 5 above)
Comment (rplopes) [inline src/index.js]: Ok, that makes sense. Thanks.

File: src/index.js (hunk 2 above)
Comment (rplopes) [inline src/index.js]: Ok. I don't have a strong opinion on this, so I'm ok with the author's reasoning.

File: src/index.js [hunk 7]
```diff
@@ -46,47 +44,58 @@ module.exports = {
...
-    'key-spacing': 'off',
-    'keyword-spacing': ['off', {
+    'key-spacing': 'error',
+    'keyword-spacing': ['error', {
       overrides: {
         case: {
           after: true
         },
+        catch: {
+          after: true,
```
Comment (rplopes) [inline src/index.js]: The default values for `after` and `before` are both `true`, so I guess all these lines can be deleted.

File: src/index.js (hunk 6 above)
Comment (ruiquelhas) [inline src/index.js]: Ok. I don't have a strong opinion on this also. I just mostly experimenting a bit.

File: src/index.js [hunk 8]
```diff
@@ -140,41 +153,44 @@ module.exports = {
...
+    'operator-linebreak': 'error',
+    'padded-blocks': ['error', 'never'],
     'prefer-arrow-callback': 'error',
     'prefer-const': 'error',
     'prefer-spread': 'error',
     'prefer-template': 'error',
-    quotes: 'off',
-    'quote-props': 'off',
+    'quote-props': ['error', 'as-needed'],
+    quotes: ['error', 'single'],
```
Comment (rplopes) [inline src/index.js]: Do we need to provide `allowTemplateLiterals: true` here to allow us to use templates where needed?

File: src/index.js [hunk 9]
```diff
@@ -122,16 +133,18 @@ module.exports = {
     'no-script-url': 'error',
     'no-self-compare': 'error',
     'no-sequences': 'error',
-    'no-shadow-restricted-names': 'error',
     'no-shadow': 'off',
-    'no-spaced-func': 'off',
+    'no-shadow-restricted-names': 'error',
+    'no-spaced-func': 'error',
     'no-sync': 'error',
-    'no-this-before-super': 'error',
```
Comment (rplopes) [inline src/index.js]: Why was this removed?

Comment (ruiquelhas) [inline src/index.js]: Part of the `recommended` config.

Comment (rplopes) [inline src/index.js]: So this isn't related to the jscs rules, right? Just about deleting some duplication? If so, I'd suggest moving that work into a separate commit, to make each commit clearer on what it's doing. Not a blocker to get this PR merged, though, just a suggestion if you agree and have the time to do it.

File: src/index.js (hunk 8 above)
Comment (ruiquelhas) [inline src/index.js]: I guess so, since we seem to use template literals even without interpolation.

File: src/index.js (hunk 7 above)
Comment (ruiquelhas) [inline src/index.js]: I guess `'error'`should also be enough.

**Comment by rplopes** [comment]
Whilst reviewing these rules, I came across a couple of them that didn't seem to make a lot of sense in my mind. I know this PR is not about challenging the existing rules, just about porting them, but after having reviewed them, this seems to be the best time to raise my concerns:
 - `disallowKeywordsInComments` - why would we want to enforce this? Aren't we using those keywords to easily track important decisions and technical debt? Is replacing those well-known keywords for generic, non-standardised alternatives a better approach?
 - `disallowMultipleLineStrings` - why would we prefer multiple string concatenations over defining a single string at once? This seems to make the code less clean and less performant.

File: src/index.js [hunk 10]
```diff
@@ -46,47 +44,35 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^[_a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z0-9]*_[A-Z-0-9]+$'],
```
Comment (rplopes) [inline src/index.js]: This still doesn't seem to allow variables such as `READY` or `GREATER_THAN_ZERO` (still only accepts one underscore).

I see the block `[A-Z-0-9]+`. Is that `-` between the Z and the 0 meant to be a `_`, or is just a typo to be deleted?

Comment (ruiquelhas) [inline src/index.js]: Typo.

File: src/index.js [hunk 11]
```diff
@@ -46,47 +44,35 @@ module.exports = {
...
     'no-cond-assign': ['error', 'always'],
     'no-confusing-arrow': 'error',
-    'no-const-assign': 'error',
-    'no-constant-condition': 'error',
     'no-div-regex': 'error',
-    'no-dupe-class-members': 'error',
     'no-duplicate-imports': 'error',
     'no-else-return': 'error',
+    'no-empty': ['error', {
+      allowEmptyCatch: true
```
Comment (franciscocardoso) [inline src/index.js]: I believe we need this to be false.

Comment (rplopes) [inline src/index.js]: If we're strictly porting the jscs rules with minimal modifications, then it makes sense to keep this, since that's the behaviour of the jscs rule: http://jscs.info/rule/disallowEmptyBlocks.

Comment (franciscocardoso) [inline src/index.js]: Indeed. Nonetheless, imho, we should fix this.

**Comment by franciscocardoso** [comment]
As discussed offline, let us add tests to ensure the behaviour of our preset.

**Comment by ruimarinho** [comment]
> disallowKeywordsInComments - why would we want to enforce this? 

We don't. It should be allowed in our case.

> disallowMultipleLineStrings - why would we prefer multiple string concatenations over defining a single string at once? This seems to make the code less clean and less performant.

This is not really a concern when using ES2015, as we don't ever do:

```
multi \
line
```

Instead, we opt for

```
`multi
line`
```

So I would disallow multiple line strings for the sake of forcing ES2015 literals in that case.

**Comment by ruimarinho** [comment]
@franciscocardoso @rplopes please merge when the PR is done.

File: src/index.js (hunk 5 above)
Comment (ruiquelhas) [inline src/index.js]: Actually, this one should really be `'error'`.

File: src/index.js [hunk 12]
```diff
@@ -46,47 +44,35 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^[_a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z0-9]*_[A-Z0-9]+$'],
```
Comment (franciscocardoso) [inline src/index.js]: The `jscs` equivalent rule _"requireCamelCaseOrUpperCaseIdentifiers"_ allows both `_camelCase` and `camelCase_`, this regex allows only `_camelCase`.

If we are aiming for a port, I would allow `camelCase_` as well. 
If we are also improving I would remove support for both cases as we don't use any of them.

/cc @nunofgs @ruimarinho @rplopes

Comment (rplopes) [inline src/index.js]: We seem to be requiring it in instances such as `this.__class__ = 'Foo';`. Or is that a property and not an id? We may also need something like this for `const _ = require('lodash');`.

Anyway, as a rule of thumb, I'm up for adapting rules to our needs and not simply blindly porting it.

File: src/index.js (hunk 11 above)
Comment (franciscocardoso) [inline src/index.js]: As discusses offline we should take advantage of this PR to fix this cases.

Please remove support to `allowEmptyCatch`.

File: src/index.js (hunk 12 above)
Comment (ruiquelhas) [inline src/index.js]: @rplopes that case should be covered by [no-underscore-dangle](http://eslint.org/docs/rules/no-underscore-dangle).

File: src/index.js (hunk 11 above)
Comment (ruiquelhas) [inline src/index.js]: Will be fixed in this PR then.

File: src/index.js [hunk 13]
```diff
@@ -140,41 +129,45 @@ module.exports = {
...
+    'no-warning-comments': 'error',
+    'no-with': 'error',
+    'object-curly-spacing': ['error', 'always'],
     'object-shorthand': 'error',
-    'one-var': 'off',
+    'one-var': ['error', 'never'],
     'operator-assignment': 'error',
-    'operator-linebreak': 'off',
-    'padded-blocks': 'off',
+    'operator-linebreak': 'error',
```
Comment (franciscocardoso) [inline src/index.js]: The default behaviour for this rule is to allow linebreak `after`. Do we really want this?
I would set this to `operator-linebreak: ["error", "none"]` instead?

WDYT @ruiquelhas ?

File: src/index.js [hunk 14]
```diff
@@ -140,41 +129,45 @@ module.exports = {
...
+    'space-unary-ops': 'error',
+    'spaced-comment': 'error',
     strict: 'off',
+    'template-curly-spacing': 'error',
     'valid-jsdoc': 'error',
     'vars-on-top': 'error',
-    yoda: 'off',
-    'wrap-iife': 'off',
-    'wrap-regex': 'error'
+    'wrap-iife': 'error',
```
Comment (franciscocardoso) [inline src/index.js]: I believe this should be `wrap-iife: ['error', 'inside']` instead?

File: src/index.js [hunk 15]
```diff
@@ -140,41 +129,45 @@ module.exports = {
...
+    'wrap-regex': 'error',
```
Comment (franciscocardoso) [inline src/index.js]: Please remove this.

File: test/fixtures/correct.js [hunk 16]
```diff
@@ -0,0 +1,271 @@
...
+noop(['foo', 'bar']);
+
+// `comma-dangle`
+noop({ bar: 'foo', foo: 'bar' });
+
+// `comma-spacing`
+noop(['bar', 'foo']);
+
+// `comma-style`
+noop({ bar: 'foo', foo: 'bar' });
```
Comment (rplopes) [inline test/fixtures/correct.js]: This is the same as in L10, right? If so, what about merging the comments, instead?

File: src/index.js [hunk 17]
```diff
@@ -46,47 +45,33 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^_$|^[a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z0-9]*_[A-Z0-9]+$'],
```
Comment (rplopes) [inline src/index.js]: This still doesn't seem to handle uppercase names with more than one underscore.

**Comment by rplopes** [comment]
I'm not a big fan of these tests' format. They look repetitive and many aren't enough to show the rule's principles that well.

I see something like this being useful if we tie it to documentation. This way, we could have some sort of coding style guide with good and bad examples, which could then be used by the tests.

I realise that might be too much to ask, though, especially considering this PR's original scope. So I leave that as a nice piece of work to do, that shouldn't block getting this merged.

Other than that, and without going through all the rules one by one again (I assume those haven't changed, at least without good reason), this looks good (just noticed a problem with the `id-match` rule).

**Comment by ruiquelhas** [comment]
@rplopes Why do you think the tests look repetitive? There's just three of them.

Anyway, I've made some changes addressing your latest comments. I urge one of you guys (including @francisco) to merge it, if there is nothing specific to add.

**Comment by rplopes** [comment]
@ruiquelhas repetitive as in many code snippets are the same across different rules being tested. At least without good documentation providing the right context, I'm not a fan of having:

```js
// `curly`.
let curly = true;

if (curly) {
  curly = false;
} else {
  curly = true;
}

// `keyword-spacing`.
let keywordSpacing = true;

if (keywordSpacing) {
  keywordSpacing = false;
} else {
  keywordSpacing = true;
}

// `no-empty`.
let noEmpty = true;

if (noEmpty) {
  noEmpty = false;
} else {
  noEmpty = true;
}

// `space-before-blocks`.
let spaceBeforeBlocks = true;

if (spaceBeforeBlocks) {
  spaceBeforeBlocks = false;
} else {
  spaceBeforeBlocks = true;
}
```

I understand where these snippets come from, and I don't have better suggestions for any of them. I think the problem isn't that they're wrong (they're not), but rather that without more explanatory context it's difficult to make anything meaningful out of them.

**Comment by ruiquelhas** [comment]
@rplopes ok. What about merging those into a single one and mention all the rules in the comment?

**Comment by rplopes** [comment]
@ruiquelhas That might be good. An alternative might be to add a bit more context to each of them, like:

```js
/**
 * `keyword-spacing`
 *
 * Keywords such as `if`, `else`, `for`, and so on must have spaces before and after,
 * for less cluttered code.
 */
let keywordSpacing = true;

if (keywordSpacing) {
  keywordSpacing = false;
} else {
  keywordSpacing = true;
}
```

Although that's probably not too far away from the proposed coding style guide I was mentioning that could be too much for this PR's scope 😛

**Comment by ruiquelhas** [comment]
@rplopes I've merged some of the rules as suggested. About the context, I'm not sure if it's actually needed, since we are referencing the specific `eslint` rule itself, so the user can just look it up on their website.

**Comment by rplopes** [comment]
@ruiquelhas Yes, the context would have to be more about our specific configuration and the reason behind it, rather than just copying the rule's description. But that's a whole documentation effort that probably not everyone agrees with, and that probably doesn't belong to this PR anyway.

So this was just my POV. I'm happy with how the current code in this PR addresses the problem this PR is set to solve 👍

**Comment by ruiquelhas** [comment]
Great, let's get this merged!

**Comment by rplopes** [comment]
@franciscocardoso does this have your LGTM as well? If so, who should do the honours?
